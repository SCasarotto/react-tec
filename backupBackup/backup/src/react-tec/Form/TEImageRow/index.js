import TEImageRow from './TEImageRow'

export default TEImageRow
