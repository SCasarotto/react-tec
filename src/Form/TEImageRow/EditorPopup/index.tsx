import React, { useState, useRef, FormEvent } from 'react'
import AvatarEditor, { AvatarEditorProps } from 'react-avatar-editor'

import { TEPopupForm, TEPopupFormProps } from '../../../Popup/TEPopupForm'
import { TELabel } from '../../TELabel'

import {
	Container,
	Row,
	RotateButtonWrapper,
	RotateButton,
	UndoRotationIcon,
	RedoRotationIcon,
} from './styledComponents'

interface Props extends Omit<TEPopupFormProps, 'onSubmit' | 'onClose'> {
	avatarEditorData: Omit<AvatarEditorProps, 'image'>
	file: string | File //Same as AvatarEditorProps image
	onSubmit(e: FormEvent<HTMLFormElement>, editor: AvatarEditor | null): void
	onCancel: TEPopupFormProps['onClose'] // This may have been a mistake in naming but at this time I don't want to make a breaking change.
}
export const EditorPopup: React.FC<Props> = (props) => {
	const [editorScale, setEditorScale] = useState(1.1)
	const [rotation, setRotation] = useState(0)
	const editor = useRef<AvatarEditor>(null)

	const {
		className = '',
		visible,
		file,
		onCancel,
		onSubmit,
		onSubmitTitle,
		avatarEditorData,
	} = props

	return (
		<TEPopupForm
			visible={visible}
			onSubmit={(e: FormEvent<HTMLFormElement>) => {
				onSubmit(e, editor.current)
			}}
			onClose={onCancel}
			onCloseTitle='Cancel'
			onSubmitTitle={onSubmitTitle}
			className={`TEEditorPopup ${className}`}
		>
			<Container className='TEImageRowPopupContainer'>
				<Row className='TEImageRowPopupRow'>
					<AvatarEditor
						ref={editor}
						image={file}
						width={300}
						height={300}
						border={20}
						// borderRadius={editorRadius}
						color={[255, 255, 255, 0.6]} // RGBA
						scale={editorScale}
						rotate={rotation}
						{...avatarEditorData}
					/>
				</Row>
				<Row className='TEImageRowPopupRow'>
					<TELabel htmlFor='scale' className='TEImageRowPopupZoomTitle'>
						Zoom
					</TELabel>
					<input
						name='scale'
						id='scale'
						type='range'
						onChange={(e) => setEditorScale(Number(e.target.value))}
						min='0.5'
						max='2.5'
						step='0.01'
						value={editorScale}
						className='TEImageRowPopupZoomSlider'
					/>
				</Row>
				<Row className='TEImageRowPopupRow'>
					<TELabel className='TEImageRowPopupRotateLabel'>Rotate</TELabel>
					<RotateButtonWrapper className='TEImageRowPopupRotateButtonWrapper'>
						<RotateButton
							className='TEImageRowPopupRotateButton'
							onClick={() => setRotation(rotation + 90)}
						>
							<UndoRotationIcon className='TEImageRowPopupRotateIcon' />
						</RotateButton>
						<RotateButton
							className='TEImageRowPopupRotateButton'
							onClick={() => setRotation(rotation - 90)}
						>
							<RedoRotationIcon className='TEImageRowPopupRotateIcon' />
						</RotateButton>
					</RotateButtonWrapper>
				</Row>
			</Container>
		</TEPopupForm>
	)
}

EditorPopup.defaultProps = {
	onSubmitTitle: 'Submit',
}
