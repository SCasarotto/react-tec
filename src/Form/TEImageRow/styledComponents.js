import styled from 'styled-components'
import { FaUpload, FaRegWindowClose } from 'react-icons/fa'

import TEButton from './../TEButton'

export const ImageRowWrapper = styled.div`
	${(props) => {
		return `
			overflow-x: auto;
			width: 100%;
			padding: 5px;
			display: flex;
			align-items: center;
		`
	}}
`
export const ImageWrapper = styled.div`
	${(props) => {
		const { theme } = props
		return `
			position: relative;
			width: 150px;
			height: 150px;
			margin-left: 0px;
			margin-right: 15px;
			border: 1px solid ${theme.lightGray};
			display: flex;
			align-items: center;
			justify-content: center;
			border-radius: 5px;
			flex-shrink: 0;
		`
	}}
`
export const Image = styled.img`
	${(props) => {
		return `
			display: block;
			width: 100%;
			object-fit: contain;
			border-radius: 5px;
		`
	}}
`
export const ClearImageButton = styled(TEButton)`
	${(props) => {
		return `
			position: absolute;
			display: flex;
			justify-content: center;
			align-items: center;
			top: 3px;
			right: 3px;
			width: auto;
			padding: 5px;
			text-align: center;
		`
	}}
`
export const ClearImageButtonIcon = styled(FaRegWindowClose)`
	${(props) => {
		return `
			color: inherit;
			display: block;
			font-size: 14px;
		`
	}}
`
export const Label = styled.label`
	${(props) => {
		const { theme, disabled } = props
		return `
			position: relative;
			display: flex;
			align-items: center;
			justify-content: center;
			width: 150px;
			height: 150px;
			flex-shrink: 0;
			padding: 30px;
			margin-left: 0px;
			margin-right: 15px;
			background-color: ${theme.white};
			cursour: pointer;
			border: 1px solid ${disabled ? theme.lighterGray : theme.lightGray};
			box-shadow: none;
			border-radius: 5px;
			transition: border-color 0.2s ease-in, box-shadow 0.2s ease-in;

			:hover {
				border: 1px solid ${disabled ? theme.lighterGray : theme.primary};
				boxShadow: ${disabled ? 'none' : `0 0 0 1px ${theme.primary} inset;`}
			}
			:active {
				border: 1px solid ${disabled ? theme.lighterGray : theme.primary};
				boxShadow: ${disabled ? 'none' : `0 0 0 1px ${theme.primary} inset;`}
			}
		`
	}}
`
export const UploadIcon = styled(FaUpload)`
	${(props) => {
		const { theme } = props
		return `
			font-size: 60px;
			color: ${theme.lightGray};
		`
	}}
`
export const Input = styled.input`
	${(props) => {
		return `
			display: none;
		`
	}}
`
export const ErrorMessage = styled.span`
	${(props) => {
		const { theme } = props
		return `
			display: block;
			padding-top: 5px;
			padding-bottom: 5px;
			font-size: 14px;
			color: ${theme.errorRed};
		`
	}}
`
