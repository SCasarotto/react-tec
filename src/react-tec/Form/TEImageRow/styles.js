import { colors } from './../../../config/styles'

export default {
	imageRowWrapper: {
		overflowX: 'auto',
		width: '100%',
		paddingTop: 5,
		paddingBottom: 5,
		paddingLeft: 5,
		paddingRight: 5,
		display: 'flex',
		alignItems: 'center',
	},
	imageWrapper: {
		position: 'relative',
		width: 150,
		height: 150,
		marginLeft: 0,
		marginRight: 15,
		borderTop: `1px solid ${colors.lightGray}`,
		borderBottom: `1px solid ${colors.lightGray}`,
		borderLeft: `1px solid ${colors.lightGray}`,
		borderRight: `1px solid ${colors.lightGray}`,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		borderRadius: 5,
		flexShrink: 0,
	},
	image: {
		display: 'block',
		width: '100%',
		// height: '100%',
		objectFit: 'contain',
		borderRadius: 5,
	},
	clearImageButton: {
		position: 'absolute',
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		top: 3,
		right: 3,
		// width: 40,
		width: 'auto',
		paddingTop: 5,
		paddingBottom: 5,
		paddingLeft: 5,
		paddingRight: 5,
		textAlign: 'center',
	},
	clearImageButtonIcon: {
		color: 'inherit',
		display: 'block',
		fontSize: 14,
	},
	label: (disabled) => {
		const styles = {
			position: 'relative',
			display: 'flex',
			alignItems: 'center',
			justifyContent: 'center',
			width: 150,
			height: 150,
			flexShrink: 0,
			padding: 30,
			marginLeft: 0,
			marginRight: 15,
			backgroundColor: colors.white,
			cursour: 'pointer',
			borderBottom: `1px solid ${colors.lightGray}`,
			borderTop: `1px solid ${colors.lightGray}`,
			borderLeft: `1px solid ${colors.lightGray}`,
			borderRight: `1px solid ${colors.lightGray}`,
			boxShadow: 'none',
			borderRadius: 5,
			transition: 'border-color 0.2s ease-in, box-shadow 0.2s ease-in',

			':hover': {
				borderBottom: `1px solid ${colors.primary}`,
				borderTop: `1px solid ${colors.primary}`,
				borderLeft: `1px solid ${colors.primary}`,
				borderRight: `1px solid ${colors.primary}`,
				boxShadow: `0 0 0 1px ${colors.primary} inset`,
			},
			':active': {
				borderBottom: `1px solid ${colors.primary}`,
				borderTop: `1px solid ${colors.primary}`,
				borderLeft: `1px solid ${colors.primary}`,
				borderRight: `1px solid ${colors.primary}`,
				boxShadow: `0 0 0 1px ${colors.primary} inset`,
			},
		}

		if (disabled) {
			styles.borderBottom = `1px solid ${colors.lighterGray}`
			styles.borderTop = `1px solid ${colors.lighterGray}`
			styles.borderLeft = `1px solid ${colors.lighterGray}`
			styles.borderRight = `1px solid ${colors.lighterGray}`

			styles.boxShadow = 'none'
			styles[':hover'].borderBottom = `1px solid ${colors.lighterGray}`
			styles[':hover'].borderTop = `1px solid ${colors.lighterGray}`
			styles[':hover'].borderLeft = `1px solid ${colors.lighterGray}`
			styles[':hover'].borderRight = `1px solid ${colors.lighterGray}`
			styles[':hover'].boxShadow = 'none'

			styles[':active'].borderBottom = `1px solid ${colors.lighterGray}`
			styles[':active'].borderTop = `1px solid ${colors.lighterGray}`
			styles[':active'].borderLeft = `1px solid ${colors.lighterGray}`
			styles[':active'].borderRight = `1px solid ${colors.lighterGray}`
			styles[':active'].boxShadow = 'none'
		}
		return styles
	},
	uploadIcon: {
		fontSize: 60,
		color: colors.lightGray,
	},
	input: { display: 'none' },
	errorMessage: {
		display: 'block',
		paddingTop: 5,
		paddingBottom: 5,
		fontSize: 14,
		color: colors.errorRed,
	},
}
