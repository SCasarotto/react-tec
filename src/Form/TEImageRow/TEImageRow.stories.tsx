import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEImageRow, TEImageRowProps } from './TEImageRow'

export default {
	title: 'Form/TEImageRow',
	component: TEImageRow,
	argTypes: {},
} as Meta

const Template: Story<TEImageRowProps> = (args) => <TEImageRow {...args} />

export const Default = Template.bind({})
Default.args = {}
