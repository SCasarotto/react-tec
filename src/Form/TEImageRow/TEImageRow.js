//
//TE Version 0.3.0
//

import React, { useState } from 'react'
import PropTypes from 'prop-types'

import TERow from './../TERow'
import TELabel from './../TELabel'
import EditorPopup from './EditorPopup'

import {
	ImageRowWrapper,
	ImageWrapper,
	Image,
	ClearImageButton,
	ClearImageButtonIcon,
	Label,
	UploadIcon,
	Input,
	ErrorMessage,
} from './styledComponents'

const TEImageRow = (props) => {
	const [editorVisible, setEditorVisible] = useState(false)
	const [file, setFile] = useState(undefined)
	const [errorData, setErrorData] = useState({ error: false, message: '' })
	const [inputKey, setInputKey] = useState(props.resetKey || 'inputKey')

	const handleClearImage = (data) => {
		props.onRemove(data)
	}
	const handleCancelEditor = () => {
		setErrorData({ error: false, message: '' })
		setFile(undefined)
		setInputKey(new Date().getTime())
	}
	const handleEditorSubmit = (editor) => {
		//TODO: Test if loading this from above will cause errors
		// const { file } = this.state
		const { onUpload } = props

		const data = { file, editor }
		onUpload(data)
			.then(() => {
				setEditorVisible(false)
				setErrorData({ error: false, message: '' })
				setFile(undefined)
				setInputKey(new Date().getTime())
			})
			.catch((error) => {
				console.log(error)
				setEditorVisible(false)
				setErrorData({ error: false, message: '' })
				setFile(undefined)
				setInputKey(new Date().getTime())
			})
	}

	const onFileChange = (e) => {
		const { pattern } = props
		const { files } = e.target

		if (!files) {
			console.error('No file provided to input.')
			return
		}

		const file = files[0]

		//Validate Files
		if (pattern && pattern instanceof RegExp) {
			if (!file.type.match(pattern)) {
				console.warn('The selected file is not an image.')
				setErrorData({ error: true, message: 'The selected file is not an image.' })
				setInputKey(new Date().getTime())
				return
			}
		}

		setErrorData({ error: false, message: '' })
		setFile(file)
		setEditorVisible(true)
	}

	const {
		//Row
		className,
		size,
		last,

		//Label
		title,

		//Input
		imgSrc,
		maxNumber,
		accept,
		disabled,

		//Both
		required,
		labelForKey,
	} = props

	const imgSrcArray = []
	if (imgSrc) {
		for (const uid in imgSrc) {
			imgSrcArray.push({ ...imgSrc[uid], uid })
		}
	}

	let labelText = title
	if (maxNumber > 1 && imgSrcArray) {
		labelText = `${title} (${imgSrcArray.length}/${maxNumber})`
	}

	return (
		<TERow size={size} last={last} className={className}>
			<TELabel
				htmlFor={labelForKey}
				required={required}
				disabled={disabled}
				className='TEImageRowTitle'
			>
				{labelText}
			</TELabel>
			<ImageRowWrapper className='TEImageRowWrapper'>
				{imgSrcArray &&
					imgSrcArray.map((imgSrcData) => {
						const { src, uid, path } = imgSrcData
						return (
							<ImageWrapper key={uid} className='TEImageRowImageWrapper'>
								<a href={src} target='_blank' rel='noopener noreferrer'>
									<Image src={src} alt={uid} className='TEImageRowImage' />
								</a>
								<ClearImageButton
									onClick={() => handleClearImage({ uid, path })}
									className='TEImageRowClearButton'
								>
									<ClearImageButtonIcon className='TEImageRowClearIcon' />
								</ClearImageButton>
							</ImageWrapper>
						)
					})}
				{(maxNumber === 0 ||
					!imgSrcArray ||
					(imgSrcArray && imgSrcArray.length < maxNumber)) &&
					!disabled && (
						<Label disabled={disabled}>
							<div>
								<UploadIcon />
							</div>
							<Input
								key={inputKey} //Used to reset input
								type='file'
								accept={accept}
								onChange={onFileChange}
							/>
						</Label>
					)}
			</ImageRowWrapper>
			{errorData && errorData.error && (
				<ErrorMessage className='TEImageRowErrorMessage'>{errorData.message}</ErrorMessage>
			)}
			<EditorPopup
				visible={editorVisible}
				file={file}
				onCancel={handleCancelEditor}
				onSubmit={handleEditorSubmit}
				className='TEImageRowPopup'
			/>
		</TERow>
	)
}

TEImageRow.propTypes = {
	//Row
	size: PropTypes.string,
	last: PropTypes.bool,

	//Input
	onUpload: PropTypes.func.isRequired,
	onRemove: PropTypes.func.isRequired,
	accept: PropTypes.string,
	// accept: PropTypes.Regexp, //Don't know protype for regex
	maxNumber: PropTypes.number,
	onChangeKey: PropTypes.string,
	disabled: PropTypes.bool,

	//Both
	labelForKey: PropTypes.string,
}

TEImageRow.defaultProps = {
	accept: 'image/*',
	pattern: /image-*/,
	maxNumber: 0,
}

export default TEImageRow
