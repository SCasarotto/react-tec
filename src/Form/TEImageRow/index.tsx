import React, { useState, ChangeEvent } from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TEFileInputProps } from 'Form/TEFileInput'
import { EditorPopup } from './EditorPopup'

import {
	ImageRowWrapper,
	ImageWrapper,
	Image,
	ClearImageButton,
	ClearImageButtonIcon,
	Label,
	UploadIcon,
	Input,
	ErrorMessage,
} from './styledComponents'

interface TEImageRowSrcObject {
	uid: string
	src: string
	// filename: string
	// fileEnding: string
	path: string
}
interface TEImageRowOnRemoveData {
	uid: string
	path: string
	index: number
}
export interface TEImageRowProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TEFileInputProps {
	labelForKey: string
	imgSrcArray: TEImageRowSrcObject[]
	maxNumber: number
	filePattern?: RegExp
	// multiple?: false
	avatarEditorData: any
	onRemove(data: TEImageRowOnRemoveData): Promise<any>
	onUpload({ file, editor }: { file?: File; editor: any }): Promise<any>
}
const TEImageRow: React.FC<TEImageRowProps> = (props) => {
	const [editorVisible, setEditorVisible] = useState(false)
	const [file, setFile] = useState<File | undefined>(undefined)
	const [errorData, setErrorData] = useState({ error: false, message: '' })
	const [inputKey, setInputKey] = useState<string | number>(
		props.resetKey || 'inputKey',
	)

	const handleClearImage = (data: TEImageRowOnRemoveData) => {
		props.onRemove(data)
	}
	const handleCancelEditor = () => {
		setErrorData({ error: false, message: '' })
		setFile(undefined)
		setInputKey(new Date().getTime())
	}
	const handleEditorSubmit = async (editor: any) => {
		try {
			const { onUpload } = props

			const data = { file, editor }
			await onUpload(data)
			setEditorVisible(false)
			setErrorData({ error: false, message: '' })
			setFile(undefined)
			setInputKey(new Date().getTime())
		} catch (e) {
			console.log(e)
			setEditorVisible(false)
			setErrorData({ error: false, message: '' })
			setFile(undefined)
			setInputKey(new Date().getTime())
		}
	}

	const onFileChange = (e: ChangeEvent<HTMLInputElement>) => {
		const { filePattern } = props
		const { files } = e.target

		if (!files) {
			console.error('No file provided to input.')
			return
		}

		const file = files[0]

		//Validate Files
		if (filePattern && filePattern instanceof RegExp) {
			if (!file.type.match(filePattern)) {
				console.warn('The selected file is not an image.')
				setErrorData({
					error: true,
					message: 'The selected file is not an image.',
				})
				setInputKey(new Date().getTime())
				return
			}
		}

		setErrorData({ error: false, message: '' })
		setFile(file)
		setEditorVisible(true)
	}

	const {
		className = '',
		rowSize,
		last,
		title,
		imgSrcArray,
		maxNumber,
		accept,
		disabled,
		required,
		labelForKey,
		avatarEditorData,
	} = props

	let labelText = title
	if (maxNumber > 1 && imgSrcArray) {
		labelText = `${title} (${imgSrcArray.length}/${maxNumber})`
	}

	return (
		<TERow
			rowSize={rowSize}
			last={last}
			className={`TEImageRow ${className}`}>
			<TELabel
				htmlFor={labelForKey}
				required={required}
				disabled={disabled}
				className="TEImageRowTitle">
				{labelText}
			</TELabel>
			<ImageRowWrapper className="TEImageRowWrapper">
				{imgSrcArray &&
					imgSrcArray.map((imgSrcData, index) => {
						const { src, uid, path } = imgSrcData
						return (
							<ImageWrapper
								key={uid}
								className="TEImageRowImageWrapper">
								<a
									href={src}
									target="_blank"
									rel="noopener noreferrer">
									<Image
										src={src}
										alt={uid}
										className="TEImageRowImage"
									/>
								</a>
								<ClearImageButton
									onClick={() =>
										handleClearImage({ uid, path, index })
									}
									className="TEImageRowClearButton">
									<ClearImageButtonIcon className="TEImageRowClearIcon" />
								</ClearImageButton>
							</ImageWrapper>
						)
					})}
				{(maxNumber === 0 ||
					!imgSrcArray ||
					(imgSrcArray && imgSrcArray.length < maxNumber)) &&
					!disabled && (
						<Label disabled={disabled}>
							<div>
								<UploadIcon />
							</div>
							<Input
								key={inputKey} //Used to reset input
								type="file"
								accept={accept}
								onChange={onFileChange}
							/>
						</Label>
					)}
			</ImageRowWrapper>
			{errorData && errorData.error && (
				<ErrorMessage className="TEImageRowErrorMessage">
					{errorData.message}
				</ErrorMessage>
			)}
			<EditorPopup
				visible={editorVisible}
				file={file}
				onCancel={handleCancelEditor}
				onSubmit={handleEditorSubmit}
				className="TEImageRowPopup"
				avatarEditorData={avatarEditorData}
			/>
		</TERow>
	)
}

TEImageRow.defaultProps = {
	accept: 'image/*',
	filePattern: /image-*/,
	maxNumber: 0,
}
