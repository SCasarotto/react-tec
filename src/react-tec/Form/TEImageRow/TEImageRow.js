//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'
import { FaUpload, FaRegWindowClose } from 'react-icons/fa'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TEButton from './../TEButton'
import EditorPopup from './EditorPopup'

import styles from './styles'

class TEImageRow extends Component {
    state = { error: false, errorMessage: '', editorVisible: false, file: undefined }

    handleClearImage = (data) => {
        this.props.onRemove(data)
    }
    handleCancelEditor = () => {
        this.setState({
            editorVisible: false,
            file: undefined,
        })
        this.refs.input.value = null
    }
    handleEditorSubmit = (editor) => {
        const { file } = this.state
        const { onUpload } = this.props

        const data = { file, editor }
        onUpload(data)
            .then(() => {
                this.setState({
                    error: false,
                    errorMessage: '',
                    editorVisible: false,
                    file: undefined,
                })

                if (this.refs.input) {
                    this.refs.input.value = null
                }
            })
            .catch((error) => {
                console.log(error)
                this.setState({
                    error: false,
                    errorMessage: '',
                    editorVisible: false,
                    file: undefined,
                })

                if (this.refs.input) {
                    this.refs.input.value = null
                }
            })
    }

    onFileChange = (e) => {
        const { pattern } = this.props
        const { files } = e.target

        if (!files) {
            console.error('No file provided to input.')
            return
        }

        const file = files[0]

        //Validate Files
        console.log(pattern)
        if (pattern && pattern instanceof RegExp) {
            if (!file.type.match(pattern)) {
                console.warn('The selected file is not an image.')
                this.setState({
                    error: true,
                    errorMessage: 'The selected file is not an image.',
                })
                this.refs.input.value = null
                return
            }
        }

        this.setState({
            error: false,
            errorMessage: '',
            editorVisible: true,
            file,
        })
    }

    render() {
        const { error, errorMessage, editorVisible, file } = this.state
        const {
            //Row
            rowStyles,
            size,
            last,

            //Label
            labelStyles,
            title,

            //Input
            imgSrc,
            // inputStyles,
            maxNumber,
            accept,
            disabled,

            //Both
            required,
            labelForKey,
        } = this.props

        const imgSrcArray = []
        if (imgSrc) {
            for (const uid in imgSrc) {
                imgSrcArray.push({ ...imgSrc[uid], uid })
            }
        }

        let labelText = title
        if (maxNumber > 1 && imgSrcArray) {
            labelText = `${title} (${imgSrcArray.length}/${maxNumber})`
        }

        return (
            <TERow size={size} last={last} style={rowStyles}>
                <TELabel
                    htmlFor={labelForKey}
                    labelText={labelText}
                    style={labelStyles}
                    required={required}
                    disabled={disabled}
                />
                <div style={styles.imageRowWrapper}>
                    {imgSrcArray &&
                        imgSrcArray.map((imgSrcData) => {
                            const { src, uid, path } = imgSrcData
                            return (
                                <div style={styles.imageWrapper} key={uid}>
                                    <a href={src} target="_blank" rel="noopener noreferrer">
                                        <img src={src} alt={uid} style={styles.image} />
                                    </a>
                                    <TEButton
                                        style={styles.clearImageButton}
                                        onClick={() => this.handleClearImage({ uid, path })}
                                    >
                                        <FaRegWindowClose style={styles.clearImageButtonIcon} />
                                    </TEButton>
                                </div>
                            )
                        })}
                    {(!imgSrcArray || (imgSrcArray && imgSrcArray.length < maxNumber)) &&
                        !disabled && (
                            <label style={styles.label(disabled)}>
                                <div style={styles.uploadIconWrapper}>
                                    <FaUpload style={styles.uploadIcon} />
                                </div>
                                <input
                                    type="file"
                                    ref="input"
                                    accept={accept}
                                    style={styles.input}
                                    onChange={this.onFileChange}
                                />
                            </label>
                        )}
                </div>
                {error && <span style={styles.errorMessage}>{errorMessage}</span>}
                <EditorPopup
                    visible={editorVisible}
                    file={file}
                    onCancel={this.handleCancelEditor}
                    onSubmit={this.handleEditorSubmit}
                />
            </TERow>
        )
    }
}

TEImageRow.propTypes = {
    //Row
    rowStyles: PropTypes.object,
    size: PropTypes.string,
    last: PropTypes.bool,

    //Input
    // inputStyles: PropTypes.object,
    onUpload: PropTypes.func.isRequired,
    onRemove: PropTypes.func.isRequired,
    accept: PropTypes.string,
    // accept: PropTypes.Regexp, //Don't know protype for regex
    maxNumber: PropTypes.number,
    onChangeKey: PropTypes.string,
    disabled: PropTypes.bool,

    //Both
    labelForKey: PropTypes.string,
}

TEImageRow.defaultProps = {
    accept: 'image/*',
    pattern: /image-*/,
    maxNumber: 1,
}

export default Radium(TEImageRow)
