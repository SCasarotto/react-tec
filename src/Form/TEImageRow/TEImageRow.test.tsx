import React from 'react'
import { render } from '@testing-library/react'
import { TEImageRow } from './TEImageRow'

describe('TEImageRow', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(
			<TEImageRow
				labelForKey='testTEImageRow'
				imgSrcArray={[]}
				onRemove={async () => {}}
				onUpload={async () => {}}
			/>,
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
