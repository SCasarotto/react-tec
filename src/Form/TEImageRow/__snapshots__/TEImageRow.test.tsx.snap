// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEImageRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c1 {
  overflow-x: auto;
  width: 100%;
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 150px;
  height: 150px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 30px;
  margin-left: 0px;
  margin-right: 15px;
  cursor: pointer;
  border: 1px solid;
  box-shadow: none;
  border-radius: 5px;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c2:hover,
.c2:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px undefined inset;
}

.c3 {
  font-size: 60px;
}

.c4 {
  display: none;
}

<div
    class="c0 TERow TEImageRow "
  >
    <div
      class="c1 TEImageRowWrapper"
    >
      <label
        class="c2"
      >
        <div>
          <svg
            class="c3"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
            />
          </svg>
        </div>
        <input
          accept="image/*"
          class="c4"
          type="file"
        />
      </label>
    </div>
  </div>
</DocumentFragment>
`;
